lib/types.js

```diff
@@ -156,8 +156,12 @@ module.exports = function(expect) {
     equal(a, b, equal) {
       return utils.checkObjectEqualityUsingType(a, b, this, equal);
     },
-    hasKey(obj, key) {
-      return key in obj;
+    hasKey(obj, key, own) {
+      if (own) {
+        return Object.prototype.hasOwnProperty.call(obj, key);
```

Comment:
Just to be clear, I guess the use of hasOwnProperty from the `Object` is to avoid having any problems with objects that might have been created using `Object.create(null)` such that their prototype chain is empty?

I'm trying to find points of reference for this new `own` situation - I guess it's a consequence of getting literally all the possible property names. I've got this niggling feeling tat this conditional should be contained entirely within "to satisfy" rather than end up in `hasKey`, but I cannot yet figure out why yet..

Reply:
> Just to be clear, I guess the use of hasOwnProperty from the `Object` is to avoid having any problems with objects that might have been created using `Object.create(null)` such that their prototype chain is empty?

Yeah, correct :+1:

Reply:
> I've got this niggling feeling tat this conditional should be contained entirely within "to satisfy" rather than end up in `hasKey`

Yeah, I feel the same way. All the `valueType` stuff that has to do with satisfying against a subtype of object is a bit weird, maybe we don't actually want it.